dev/build_release.py

```diff
@@ -222,6 +222,9 @@ def __debian_build(cls, name, options, build_number, gradle_root):
       '-PbintrayUser="{user}"'.format(user=bintray_user)
     ]
 
+    if options.debian_gradle_init_file:
```

Comment:
This still looks awkward because you initialize this in two different places and the parameter is specified in build_release but is based on a file in a different module entirely.

Why not instead have a parameter for the "maven_custom_initialization_file", which is empty by default. Then here check if there was one supplied or not. If so then use it. Otherwise generate a temporary file containing a static string (the body of the init script below) and use that instead. Create a local variable containing the name of the file to delete (empty by default) then wrap the command execution in a try/finally where you check and delete in the finally block.

There is no change to generate_bom or build_prevalidate. This assumes that there is only one context for executing maven. You encapsulate the maven configuration here in the same module that runs maven, and you localize the use of the configuration right where its used.

Reply:
SGTM.